### DISCUSSION TITLE
piloter des "light" en plus des Switch et des input_boolean

### DISCUSSION THREAD

**Discussion by MiGaNuTs** [jmcollin78/versatile_thermostat#503] [Ideas]
Bonjour.
J'ai un problème, c'est que je n'aime pas trop les relais et autres contacts secs. Je préfère largement les relais statiques, thyristors et autres interrupteurs électroniques. Ça fait moins de bruit, et ça supporte beaucoup plus de cycles de commutation avant de rendre l'âme.
Du coup je me suis fabriqué mes propres relais pour commander mes radiateur, a base d'ESP sous Tasmota et de relais statiques largement surdimensionnés par rapport à mes radiateurs pour être certain qu'ils vont durer longtemps.
Du coup j'en profite pour les piloter en pwm, ce qui fait que HA les voit comme des lampes.
Pour les utiliser avec Versatile Thermostat je suis obligé de créer un helper input_boolean vide, et un script période qui va lire le % de puissance demandée par le thermostat, le multiplier par 2.55 puis le recopier dans la consigne de ma "lampe". (Et de l'éteindre si 0, mais l'allumer sinon)
Ça serait plus simple pour si versatile thermostat acceptait de piloter directement une entité de type lampe et d'y mettre la consigne d'éclairage a l'échelle.
Est ce que la modification est est facile a faire ?

**Comment by jmcollin78** [comment]
Hello @MiGaNuTs ,

You talk about a new type of VTherm. Even if it is not so hard to do, it's a big change. And because, it is a very specific use case, I will not do it.

What you can try:
1. create a virtual `number` entity that will finally controls you light,
2. create a VTherm `over_valve` and configure the `number` as underlying device. Set min and max accordingly to your need.

This should work out the box. Let us know, this can be interesting for others (in English please).

**Reply by MiGaNuTs** [reply]
Thanks for the trick. For the moment i will continue with my automations, but why not trying this later.
